src/puppetlabs/puppetdb/query_eng.clj

```diff
@@ -101,55 +105,77 @@
 
       (munge-fn version url-prefix))))
 
-(defn stream-query-result
+(def query-options-schema
+  {:scf-read-db s/Any
+   (s/optional-key :url-prefix) String
+   (s/optional-key :warn-experimental) Boolean
+   (s/optional-key :pretty-print) Boolean})
+
+(pls/defn-validated stream-query-result
   "Given a query, and database connection, return a Ring response with the query
    results."
-  ([entity version query paging-options db url-prefix]
+  ([version query paging-options options]
    ;; We default to doall because tests need this for the most part
-   (stream-query-result entity version query paging-options db url-prefix doall))
-  ([entity version query paging-options db url-prefix row-fn]
-   (let [munge-fn (get-munge-fn entity version paging-options url-prefix)]
-     (jdbc/with-transacted-connection db
+   (stream-query-result version query paging-options options doall))
+  ([version :- s/Keyword
+    query
+    paging-options
+    options :- query-options-schema
+    row-fn]
+   (let [{:keys [scf-read-db url-prefix warn-experimental pretty-print]
+          :or {warn-experimental true
+               pretty-print false
+               url-prefix "/pdb"}} options
```

Comment:
Well, we have it defaulted in config, so on second thoughts I'll just stop defaulting it and make it mandatory. The decision to do this anyway was fairly arbitrary, it just helped 2-3 test cases, but not really necessary. I'm going to remove the defaulting and make it mandatory - since it did help detect a few cases I noticed where we weren't setting it.